syntax = "proto3";

package todo_list;

message TodoItem 
{
  string id = 1;
  string title = 2;
  string note = 3;
  string todo_list_id = 4;
  bool completed = 5;
  string created_date = 6;
  string modified_date = 7;
  int32 version = 8;
}

message AddTodoRequest 
{
	string title = 2;
	string description = 3;
}

message UpdateTodoRequest  
{
	string id = 1;
	string title = 2;
	string description = 3;
}

message GetTodoListRequest 
{
	string id = 1;
}

message TodoList 
{
	string id = 1;
	string title = 2;
	string description = 3;
	string created_date = 4;
	string modified_date = 5;
	int32 version = 6;
	repeated TodoItem todo_item = 7;
}

message DeleteTodoRequest 
{
	string id = 1;
}

message AddTodoListResponse 
{
	string id = 1;
	int32 status = 2;
	optional string message = 3;
}

message UpdateTodoListResponse 
{
	int32 status = 1;
	optional string message = 2;
}

message GetTodoListResponse 
{
	TodoList todo_list = 1;
	int32 status = 2;
	optional string message = 3;
}

message DeleteTodoListResponse 
{
	int32 status = 1;
	optional string message = 2;
}

service TodoListService 
{
	rpc AddTodoList (AddTodoRequest) returns ( AddTodoListResponse ) {}
	rpc UpdateTodoList (UpdateTodoRequest) returns ( UpdateTodoListResponse ) {}
	rpc GetTodoList (GetTodoListRequest) returns ( GetTodoListResponse ) {}
	rpc DeleteTodoList (DeleteTodoRequest) returns ( DeleteTodoListResponse ) {}
}